ibis/expr/types/relations.py

```diff
@@ -1993,31 +2000,34 @@ def select(
         import ibis.expr.analysis as an
         from ibis.selectors import Selector
 
-        exprs = [
-            e
-            for expr in exprs
-            for e in (
-                expr.expand(self)
-                if isinstance(expr, Selector)
-                else map(self._ensure_expr, util.promote_list(expr))
-            )
-        ]
-        exprs.extend(
+        new_exprs = []
+
+        for expr in exprs:
+            if isinstance(expr, Selector):
+                new_exprs.extend(expr.expand(self))
```

Comment:
`_ensure_expr` handles selector expansion, I *think* this case could be dropped and handled by the not-mapping branch below?

Reply:
Unfortunately it won't work because `expr.expand(self)` returns a `list` and so `map(self._ensure_expr, ...)` returns an iterable of `list`s and that isn't supported by `Projector`.